### DISCUSSION TITLE
Validating Method Parameters

### DISCUSSION THREAD

**Discussion by dreckner** [MarimerLLC/csla#1731] [Questions]
I have a static method on a ReadOnlyList that takes a string as a parameter.  The search string accepts a full name, email address, transaction number, a transaction number range, a single date and a date range. What is the best way to protect the parameter from unexpected characters?

**Comment by GreatBarrier86** [comment]
Given that Microsoft's own code throws the ArgumentNullException where appropriate, I'd imagine that just validating the parameters with either a String.Contains or Regex.IsMatch would be a reasonable implementation.

In principle, that would probably be similar to the logic that would satisfy this design warning [here](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1062-validate-arguments-of-public-methods?view=vs-2015&redirectedfrom=MSDN)
